src/main/webapp/app/overview/course-registration-selector/course-registration-selector.component.ts

```diff
@@ -31,8 +31,14 @@ export class CourseRegistrationSelectorComponent implements OnInit {
             this.profileService.getProfileInfo().subscribe((profileInfo) => {
                 if (profileInfo) {
                     if (profileInfo.allowedCourseRegistrationUsernamePattern) {
-                        const regex = new RegExp(profileInfo.allowedCourseRegistrationUsernamePattern);
-                        this.userIsAllowedToRegister = !!user!.login!.match(regex);
+                        // we want to test for a full match, so wrap the regex in ^ and $
+                        let regex = profileInfo.allowedCourseRegistrationUsernamePattern;
+                        if (!regex.startsWith('^') && !regex.endsWith('$')) {
```

Comment:
this can lead to a problem if a regex is specified with only a start or only a regex end e.g. `^[a-zA-Z0-9_\-\.\+]+@(gmail.com)` or `[a-zA-Z0-9_\-\.\+]+@(gmail.com)$`

Reply:
That should be a `||` instead of `&&`. Will fix, thanks.

Reply:
never mind I think it is correct just got it wrong, but one question: what happens to the case when we do start only with `^` or end with `$` should it then also be wrapped?